src/ucp/core/ucp_mm.c

```diff
@@ -576,6 +576,16 @@ ucp_memh_init_from_parent(ucp_mem_h memh, ucp_md_map_t parent_md_map)
     }
 }
 
+static void
+ucp_memh_check_odp_memory_types(ucp_context_h context,
```

Comment:
Maybe rename it?
I'd expect a `check` function to be a boolean predicate or to `assert()` something, rather than apply config.